### PR TITLE
[Greenfield] Document store API

### PR DESCRIPTION
**Pull request by NicolasDorier**
This PR is fixing a number of issue over default in the store API.

It is also documenting all the fields.

### CODE REVIEW

File: BTCPayServer/Controllers/GreenField/StoresController.cs [hunk 1]
```diff
@@ -128,7 +128,7 @@ private static Client.Models.StoreData FromModel(Data.StoreData data)
                 CustomLogo = storeBlob.CustomLogo,
                 CustomCSS = storeBlob.CustomCSS,
                 HtmlTitle = storeBlob.HtmlTitle,
-                AnyoneCanInvoice = storeBlob.AnyoneCanInvoice,
+                AnyoneCanCreateInvoice = storeBlob.AnyoneCanInvoice,
```
Comment (Kukks) [inline BTCPayServer/Controllers/GreenField/StoresController.cs:131]: If we're renaming this, maybe `PublicInvoiceEndpointEnabled`?

File: BTCPayServer/wwwroot/swagger/v1/swagger.template.stores.json [hunk 2]
```diff
@@ -306,58 +304,79 @@
...
+                        "description": "If true, the checkout page will ask to enter an email address before accessing payment information."
                     },
                     "networkFeeMode": {
                         "$ref": "#/components/schemas/NetworkFeeMode"
                     },
                     "payJoinEnabled": {
-                        "type": "boolean"
+                        "type": "boolean",
+                        "default": false,
+                        "description": "If true, payjoin will be proposed in the checkout page if possible. ([More information](https://docs.btcpayserver.org/Payjoin/#btcpay-server-payjoin-guide))"
```
Comment (Kukks) [inline BTCPayServer/wwwroot/swagger/v1/swagger.template.stores.json:374]: does that link actually work in redoc?

Comment (NicolasDorier) [reply BTCPayServer/wwwroot/swagger/v1/swagger.template.stores.json:374]: it does! redoc is awesome

File: BTCPayServer/Controllers/GreenField/StoresController.cs (hunk 1 above)
Comment (NicolasDorier) [reply BTCPayServer/Controllers/GreenField/StoresController.cs:131]: I find it more confusing.

**Pull request merged**